### PR TITLE
[NEW CONTROL] GR1 V2 - All Cloud User Accounts MFA Conditional Access Policy Check

### PR DESCRIPTION
**Pull request by alalvi00**
Created a new control in GR1 based on GR3 control but with more flexibility towards the conditions of CAPs

More conditions will be added later on as we work on this control

## Overview/Summary

Created a new control in Guardrail 1 called Check-CloudAccountsMFA which is based on the control in Guardrail 3. The new control allows more conditions to be checked for the Conditional Access Policies giving a more granular level of MFA check.

## This PR fixes/adds/changes/removes

1. Adds a new control in modules.json
2. A new module in GR1 and psmodules zip

### Breaking Changes

N/A

## Testing Evidence

No testing evidence since it is a copy of GR3 control with the conditions changed.

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/ssc-spc-ccoe-cei/azure-guardrails-solution-accelerator/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/ssc-spc-ccoe-cei/azure-guardrails-solution-accelerator/issues)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/ssc-spc-ccoe-cei/azure-guardrails-solution-accelerator/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.
- [x] Ensure PowerShell module versions have been updated (manually or with the ./tools/Update-ModuleVersions.ps1 script)

### CODE REVIEW

File: setup/modules.json
Comment (dutt0) [inline setup/modules.json:1]: Profiles array value is not correct.

**Review by dutt0 — CHANGES_REQUESTED** [review]
1. Please follow our naming convention for PR title
2. Where is the ItemName for this control? The GR-ComplianceChecks-Msgs are not included in the PR

File: setup/modules.json
Comment (alalvi00) [reply setup/modules.json:1]: Not sure about this, could you provide the right values?

**Comment by alalvi00** [comment]
> 1. Please follow our naming convention for PR title
> 2. Where is the ItemName for this control? The GR-ComplianceChecks-Msgs are not included in the PR


1) Fixed
2) New compliance messages are not needed since it uses the messages from the previous control

File: setup/modules.json
Comment (dutt0) [reply setup/modules.json:1]: For guardrail1, [1, 2, 3, 4, 5, 6]

**Comment by dutt0** [comment]
> > 1. Please follow our naming convention for PR title
> > 2. Where is the ItemName for this control? The GR-ComplianceChecks-Msgs are not included in the PR
> 
> 1. Fixed
> 2. New compliance messages are not needed since it uses the messages from the previous control

2. Please update the ItemName in English and French (in the ticket) for this GR control @alalvi00

File: setup/modules.json [hunk 1]
```diff
@@ -24,6 +24,21 @@
       }
     ]
   },
+  {
+    "ModuleName": "Check-CloudAccountsMFA",
+    "Control": "Guardrails1",
+    "ModuleType": "Builtin",
+    "Status": "Enabled",
+    "Required": "True",
+    "Profiles": [2, 3, 4, 5, 6],
```
Comment (singhgss) [inline setup/modules.json:33]: Think @dutt0  was referring to this profile array in modules.json

**Review by singhgss — APPROVED** [review]
Approved with suggested changes in module.json

**Review by dutt0 — CHANGES_REQUESTED** [review]
Since this is an enhancement on an existing control and new file added to this PR,  we need to remove all the relevant files associated with the existing GR3 'Multi-Factor authentication required for all users by Conditional Access Policy' control

**Review by dutt0 — APPROVED** [review]
LGTM

**Pull request merged**